FWCore/Utilities/src/MallocOpts.cc

```diff
@@ -162,6 +162,7 @@ namespace edm
     error_message_.clear();
     changed_ = false;
 
+#ifndef __SANITIZE_ADDRESS__
```

Comment:
Compile time + run-time library. The main executable (cmsRun) is statically linked to ASan run-time, which initializes required structures and brings all symbols. At compile time it adds and replaces functions calls with special functions in ASan run-time which do all accounting.